### PR TITLE
[dynamo 3.11] support new binary ops

### PR DESCRIPTION
**Pull request by williamwen42**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #94243
* #94102
* #94101
* #94100
* #94099
* #94098
* #93988
* __->__ #93987
* #93986

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: torch/_dynamo/symbolic_convert.py [hunk 1]
```diff
@@ -1491,6 +1493,16 @@ def MATCH_KEYS(self, inst):
     def RESUME(self, inst):
         pass
 
+    def BINARY_OP(self, inst):
+        # linter complains about dis missing _nb_ops in py < 3.11
```
Comment (albanD) [inline torch/_dynamo/symbolic_convert.py:1497]: Is it acutally there? Should we just silence the linter to make this work on all python versions?

Comment (williamwen42) [reply torch/_dynamo/symbolic_convert.py:1497]: `_nb_ops` is not present in python < 3.11

**Review by mlazos — APPROVED** [review]

File: torch/_dynamo/symbolic_convert.py (hunk 1 above)
Comment (albanD) [reply torch/_dynamo/symbolic_convert.py:1497]: Ho so the comment needs updating right? The attribute is not there for py < 3.11. It has nothing to do with the linter.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**